include/zephyr/sys/byteorder.h

```diff
@@ -719,4 +721,120 @@ static inline void sys_mem_swap(void *buf, size_t length)
 	}
 }
 
+/**
+ *  @brief Convert buffer from little-endian to host endianness.
```

Comment:
Please properly document all the `@param`s of the functions you're introducing

Reply:
Why didn't CI catch this? :o